File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        ListNode curr = head;
        int size = 0;
        while(curr != null){
            curr = curr.next;
            size++;
        }
        
        curr = head;
        int mid = size / 2;
        for(int i = 0; i < mid; i++){
            curr = curr.next;
        }
        
        return curr;
    }
}

File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        curr = head
        size = 0
        while curr:
            curr = curr.next
            size += 1
            
        curr = head
        for _ in range(size // 2):
            curr = curr.next
            
        return curr